crates/starknet-devnet-server/src/builder.rs

```diff
@@ -103,6 +110,52 @@ impl<TJsonRpcHandler: RpcHandler, THttpApiHandler: Clone + Send + Sync + 'static
                     .allow_methods(vec![Method::GET, Method::POST]),
             );
 
+        if config.log_request {
+            svc = svc.layer(axum::middleware::from_fn(request_logging_middleware));
+        }
+
         Ok(Server::try_bind(&self.address)?.serve(svc.into_make_service()))
     }
 }
+
+async fn request_logging_middleware(
+    request: Request<hyper::Body>,
+    next: Next<hyper::Body>,
+) -> Result<impl IntoResponse, Response> {
+    let (parts, body) = request.into_parts();
+
+    let body = log_body(body).await?;
```

Comment:
Would it make sense to also log the request path? Currently the output looks like this after sending a few dummy 404 requests:
```
2024-05-03T07:49:14.607496Z  INFO starknet_devnet: Starknet Devnet listening on 127.0.0.1:5050
2024-05-03T07:49:24.512071Z  INFO starknet_devnet_server::builder: 
2024-05-03T07:49:24.512189Z  INFO starknet_devnet_server::builder: 
2024-05-03T07:49:27.157357Z  INFO starknet_devnet_server::builder: 
2024-05-03T07:49:27.157427Z  INFO starknet_devnet_server::builder: 
2024-05-03T07:49:36.963082Z  INFO starknet_devnet_server::builder: {}
2024-05-03T07:49:36.963151Z  INFO starknet_devnet_server::builder: 
```

It is not clear which requests the logs correspond to.

Reply:
[d4fb2f1](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/455/commits/d4fb2f11a928cb684cde689929dfe83188b12c81)